### PR TITLE
 Monkeycli

### PR DESCRIPTION
**Pull request by niktrix**
npm run monkey fill 1000

### CODE REVIEW

**Comment by niktrix** [comment]
#29

File: bin/monkey-cli.ts [hunk 1]
```diff
@@ -91,21 +112,64 @@ function ethcall(txParams: Txp): Promise<any> {
...
+
 async function fillAccountsWithEther(txParams: Txp): Promise<any> {
+  let gp = '0x7B0C'
   for (const account of accounts) {
     txParams.to = account.address
     txParams.value = '0x2000000000000000'
     const privateKey = Buffer.from(from.key, 'hex')
     try {
-      ora.info(`Sending: ${txParams.value} wei to ${txParams.to}`)
+      const r = await estimategas(txParams)
```
Comment (aldoborrero) [inline bin/monkey-cli.ts]: Change to estimateGas

File: bin/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# monkeycli
+Ethereum utility to Generate random transactions
```
Comment (aldoborrero) [inline bin/README.md]: Add space between title and paragraph

File: bin/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+ 
+### Usage
+
+#### Step 1 
+Deploy ERC20 token contract
+```sh
```
Comment (aldoborrero) [inline bin/README.md]: Add space between paragraph and '''sh

File: bin/README.md [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+$ npm run monkey deploy
+```
+#### Step 2     
+Check if contract is deployed and get contract address
```
Comment (aldoborrero) [inline bin/README.md]: Same here

File: bin/README.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
+# monkeycli
+Ethereum utility to Generate random transactions
+ 
+### Usage
+
+#### Step 1 
+Deploy ERC20 token contract
+```sh
+$ npm run monkey deploy
+```
```
Comment (aldoborrero) [inline bin/README.md]: Same here

File: bin/README.md [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+#### Step 2     
+Check if contract is deployed and get contract address 
+npm run monkey txdetail <txhash>
+```sh
```
Comment (aldoborrero) [inline bin/README.md]: Same here

File: bin/monkey-cli.ts [hunk 7]
```diff
@@ -8,10 +8,14 @@ import { bufferToHex, generateAddress, toBuffer } from 'ethereumjs-util'
 import Ora from 'ora'
 import * as utils from 'web3-utils'
 
+import { resolve } from 'dns'
```
Comment (aldoborrero) [inline bin/monkey-cli.ts]: Is this module installed or added to package.json?

File: bin/README.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+$ npm run monkey txdetail 0xe17c0b51510dac4aa8f06f51db040d5e9f0c5af4371b875cdd8d964ee34fc554
+```
+
+####  step 3 
+
+start generating random txs
+npm run monkey start <contractaddress>
```
Comment (aldoborrero) [inline bin/README.md]: Add space and '''sh

File: bin/monkey-cli.ts [hunk 9]
```diff
@@ -74,12 +79,28 @@ function send(txP: Txp, privateKey: Buffer): Promise<Result> {
   })
 }
 
-function ethcall(txParams: Txp): Promise<any> {
+function estimategas(txParams: Txp): Promise<any> {
```
Comment (aldoborrero) [inline bin/monkey-cli.ts]: Change to estimateGas

File: bin/monkey-cli.ts (hunk 7 above)
Comment (niktrix) [reply bin/monkey-cli.ts]: not used, removed

**Pull request merged**